doc/source/user/numpy-for-matlab-users.rst

```diff
@@ -25,165 +25,50 @@ Some Key Differences
+       creates two-dimensional arrays of double precision floating point numbers.
+       Operations on the 2D instances of these arrays are modeled on
+       matrix operations in linear algebra. 
+     - In NumPy, the basic type is a multidimensional ``array``. By default,
+       NumPy creates 0-D or 1-D arrays for inputs of numbers or lists,
+       respectively.  Operations on these arrays in all dimensionalities
+       including 2D are element-wise operations.  One needs to use specific
+       functions for linear algebra (though for matrix multiplication, one can
+       use the ``@`` operator in Python 3.5 and above).
+
+   * - MATLAB uses 1 (one) based indexing. The initial element of a
        sequence is found using a(1).
        :ref:`See note INDEXING <numpy-for-matlab-users.notes>`
      - Python uses 0 (zero) based indexing. The initial element of a
        sequence is found using a[0].
 
-   * - MATLAB®'s scripting language was created for doing linear algebra.
-       The syntax for basic matrix operations is nice and clean, but the API
-       for adding GUIs and making full-fledged applications is more or less
-       an afterthought.
-     - NumPy is  based on Python, which was designed from the outset to be
-       an excellent general-purpose programming language.  While Matlab's
+   * - MATLAB's scripting language was created for doing linear algebra so the
        syntax for some array manipulations is more compact than
-       NumPy's, NumPy (by virtue of being an add-on to Python) can do many
-       things that Matlab just cannot, for instance dealing properly with
-       stacks of matrices.
-
-   * - In MATLAB®, arrays have pass-by-value semantics, with a lazy
-       copy-on-write scheme to prevent actually creating copies until they
+       NumPy's. The syntax for basic matrix operations is nice and clean, but the
+       API for adding GUIs and making full-fledged applications is more or less
+       an afterthought.
+     - NumPy is  based on Python, which was designed from the outset to be an
+       excellent general-purpose programming language.  The advantage to NumPy
+       is access to Python libraries including: `SciPy
+       <https://www.scipy.org/>`_, `Matplotlib <https://matplotlib.org/>`_,
+       `Pandas <https://pandas.pydata.org/>`_, `OpenCV <https://opencv.org/>`_,
+       and more. In addition to Python-specific packages, much `scientific
+       computing software
+       <https://en.wikipedia.org/wiki/List_of_Python_software#Embedded_as_a_scripting_language>`_
+       use Python as a scripting language. 
+
+   * - In MATLAB, arrays have pass-by-value semantics, with a lazy
+       copy-on-write scheme to prevent creating copies until they
        are actually needed.  Slice operations copy parts of the array.
-     - In NumPy arrays have pass-by-reference semantics.  Slice operations
+     - In NumPy, arrays have pass-by-reference semantics.  Slice operations
        are views into an array.
```

Comment:
We should qualify this a little  (perhaps "arrays **often** have"). Slices are indeed views, but some operations return copies.

Reply:
@eric-wieser "Pass by reference" isn't mine.

Reply:
Does this need a rewrite? How about:

`MATLAB array slicing uses pass-by-value semantics, with a lazy copy-on-write scheme to prevent creating copies until they are needed. Slice operation copy parts of the array.`

`NumPy array slicing uses pass-by-reference, that does not copy the arguments. Slice operations are views into an array.`

Reply:
Updated to the rewrite in a3b1c6e303d6a1261b6068f4d1e406c94d178e7a.